linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -0,0 +1,350 @@
+# threadpool_linux
+
+## Overview
+
+`threadpool_linux` is the Linux-specific implementation of the `threadpool` module. It receives an execution engine at creation time in order to executes scheduled asynchronous work items after been created. `threadpool_linux` supports timer functions to execute at a regular interval.
+
+## Design
+
+`threadpool_linux` provides the Linux implementation of `threadpool`. The `threadpool` object starts a predefined number of threads that are used to execute the work items. The `threadpool` unit tracks all the work items which need to be executed in an task array. All incoming tasks are being added in `threadpool_schedule_work` and scheduled in `threadpool_work_func`. 
```

Comment:
>`threadpool` [](http://example.com/codeflow?start=56&length=12)

"of the `threadpool` PAL API." sounds better I think

Reply:
done